### PR TITLE
[centipede][base-builder] Properly delete bazel cache

### LINKED ISSUE #8468: Centipede is bloating our image sizes.
It reinstalls clang? and adds 1.7GB to our image size, which slows down every CFL/CIFuzz user.

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Fixes #8468

### CODE REVIEW

**Review by Alan32Liu — APPROVED** [review]
LGTM. Would you prefer me to address the TODO in another PR?
It is simply execute `rm -rfv !("centipede"|"libcentipede_runner.pic.a")` in `/src/centipede/bazel-bin/`.

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine centipede

**Comment by Alan32Liu** [comment]
The previous trial build does not seem to work as it failed due to `AESTStep #1: INFO:root:Trial build not requested.`
Requesting it again.

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine centipede

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine centipede

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine centipede

File: infra/base-images/base-builder/precompile_centipede [hunk 1]
```diff
@@ -36,4 +36,11 @@ echo 'Removing extra stuff leftover to avoid bloating image.'
 
 rm -rf /clang-*.tgz /clang
 
+BAZEL_BIN_REAL_DIR=$(readlink -f $SRC/centipede/bazel-bin)
+rm -rf $SRC/centipede/bazel-bin
+mkdir -p $SRC/centipede/bazel-bin
+mv $BAZEL_BIN_REAL_DIR/centipede $SRC/centipede/bazel-bin/
+mv $BAZEL_BIN_REAL_DIR/libcentipede_runner.pic.a $SRC/centipede/bazel-bin/
```
Comment (oliverchang) [inline infra/base-images/base-builder/precompile_centipede:43]: nit: you can just do one line for both: 

```
mv $BAZEL_BIN_REAL_DIR/{centipede,libcentipede_runner.pic.a} $SRC/centipede/bazel-bin/
```

**Review by oliverchang — APPROVED** [review]

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine centipede

**Comment by Alan32Liu** [comment]
/gcbrun trial_build.py all --sanitizer address --fuzzing-engine centipede

**Pull request merged**